docs/examples/DataSet/Performing-measurements-using-qcodes-parameters-and-dataset.ipynb

```diff
@@ -1527,6 +1528,79 @@
     "ax, cbax = plot_dataset(dataset)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Specifying shape of measurement\n",
+    "As the context manager allows you to generate data in any shape with the only restriction being that you supply values for both dependent and independent parameters together it cannot know if the date is to be measured on a grid and as a consequence the Numpy array of data loaded from the dataset may not be in the shape that you expect. `plot_dataset` and `get_data_as_pandas_dataframe`  contains logic that can detect the shape of the data measured at load time. However, if you know the shape of the measurement that you are going to perform up front you may chose to specify it before initializing the measurement. \n",
+    "\n",
+    "`dataset.get_parameter_data` and `dataset.cache.data` automatically makes use of this information to return shaped data when loaded from the database.\n",
```

Comment:
i believe it is out of scope for this PR to rework plot_dataset to rely on shape metadata whenever possible, right? because at the moment it ALWAYS guesses because it relies on get_data_by_id which in turn does rely on get_parameter_data however it explicitly flattens any shapes, see here https://github.com/qcodes/Qcodes/blob/master/qcodes/dataset/data_export.py#L90

Reply:
Yes that should indeed be fixed but in a different pr. We should be able to clean up that code significnantly (and perhaps even share the guess code between plottr and qcodes)